src/details/ArborX_AttachIndices.hpp

```diff
@@ -78,6 +78,9 @@ struct ArborX::AccessTraits<ArborX::Experimental::AttachIndices<Values, Index>,
   }
   KOKKOS_FUNCTION static auto get(Self const &self, int i)
   {
+#if defined(KOKKOS_COMPILER_NVCC)
+#pragma nv_diag_suppress 20014
```

Comment:
What is the 20014 warning?

Reply:
Calling __host__ function from __host__ __device__

Somehow, simple casting `(Index)i` results in this warning.

Reply:
I would expect the warning to go away if you mark the `CustomFunction` constructor as `KOKKOS_FUNCTION`:
```
KOKKOS_FUNCTION CustomIndex(int i) { index = i; }
```

Reply:
Good catch. Not sure why I thought `Index` was `int` here, when it clearly is not.